vortex-array/src/array/implementation.rs

```diff
@@ -116,9 +117,8 @@ impl<A: ArrayImpl + 'static> Array for A {
 
         // We know that constant array don't need stats propagation, so we can avoid the overhead of
         // computing derived stats and merging them in.
-        // TODO(ngates): skip the is_constant check here, it can force an expensive compute.
         // TODO(ngates): provide a means to slice an array _without_ propagating stats.
-        let derived_stats = (!self.is_constant()).then(|| {
+        let derived_stats = self.is_encoding(ConstantEncoding.id()).then(|| {
```

Comment:
```suggestion
        let derived_stats = (!self.is_encoding(ConstantEncoding.id())).then(|| {
```
You want the not still

Reply:
c&p